Backpack-SwiftUI/ImageGallerySlideshow/Classes/BPKImageGallerySlideshow.swift

```diff
@@ -0,0 +1,162 @@
+                ContentView(
+                    images: images,
+                    closeAccessibilityLabel: closeAccessibilityLabel,
+                    onCloseTapped: onCloseTapped,
+                    currentIndex: $currentIndex
+                )
+            })
+    }
+    
+    struct ContentView: View {
+        let images: [BPKImageGalleryImage<ImageView>]
+        let closeAccessibilityLabel: String
+        let onCloseTapped: () -> Void
+        @Binding var currentIndex: Int
+        
+        var body: some View {
+            GeometryReader { proxy in
+                VStack(alignment: .leading, spacing: .xl) {
+                    header
+                    
+                    ZStack(alignment: .bottom) {
+                        InternalCarouselWrapper(
+                            images: images.map { $0.content() },
+                            pageIndicatorVisibility: .hidden,
+                            currentIndex: $currentIndex
+                        )
+                        BPKBadge("\(currentIndex + 1)/\(images.count)")
+                            .badgeStyle(.strong)
+                            .padding(.bottom, 20)
+                    }
+                    .frame(width: proxy.size.width, height: proxy.size.width)
+                    
+                    footer
+                }
+            }
+            .background(Color(.canvasContrastColor))
+        }
+        
+        private var header: some View {
+            Button(action: onCloseTapped, label: {
+                BPKIconView(.close, size: .large)
+                    .foregroundColor(.textPrimaryColor)
+                    .padding(.sm)
+            })
+            .accessibilityLabel(closeAccessibilityLabel)
+            .padding(.leading, .base)
+        }
+        
+        private var footer: some View {
+            VStack(spacing: .xl) {
+                BPKPageIndicator(
+                    currentIndex: $currentIndex,
+                    totalIndicators: .constant(images.count)
+                )
+                
+                VStack(spacing: .md) {
+                    BPKText(descriptionText, style: .caption)
+                        .lineLimit(nil)
+                    
+                    if let credit = images[currentIndex].credit {
+                        HStack(spacing: .sm) {
+                            BPKIconView(.camera)
+                            BPKText(credit, style: .caption)
+                        }
+                    }
+                }
+                .accessibilityElement(children: .ignore)
+                .accessibilityLabel(descriptionAccessibilityText)
+                .padding(.horizontal, .lg)
+            }
+        }
+        
+        private var descriptionAccessibilityText: String {
+            let currentImage = images[currentIndex]
+            return "\(currentImage.title). \(currentImage.description ?? ""). \(currentImage.credit ?? "")"
+        }
```

Comment:
This is prone to failure, if we use the suggestion above this is not needed :D

Reply:
Sounds good yeah!